dynamodb-stream-reader/src/test/java/no/unit/nva/dynamodb/DynamoDBStreamHandlerTest.java

```diff
@@ -0,0 +1,128 @@
+        doReturn(ELASTICSEARCH_ENDPOINT_INDEX).when(environment)
+                .readEnv(DynamoDBStreamHandler.ELASTICSEARCH_ENDPOINT_INDEX_KEY);
+        doReturn(ELASTICSEARCH_ENDPOINT_API_SCHEME).when(environment)
+                .readEnv(ElasticSearchRestClient.ELASTICSEARCH_ENDPOINT_API_SCHEME_KEY);
+
+        doReturn(TARGET_SERVICE_URL).when(environment)
+                .readEnv(DynamoDBStreamHandler.TARGET_SERVICE_URL_KEY);
+        doReturn(ELASTICSEARCH_ENDPOINT_INDEX).when(environment)
+                .readEnv(DynamoDBStreamHandler.ELASTICSEARCH_ENDPOINT_INDEX_KEY);
+
+        HttpResponse<String> successResponse = mock(HttpResponse.class);
+        doReturn(successResponse).when(httpClient).send(any(), any());
+        elasticSearchRestClient = new ElasticSearchRestClient(httpClient, environment);
+        handler = new DynamoDBStreamHandler(elasticSearchRestClient, environment);
+    }
+
+    @Test
+    @DisplayName("testCreateHandlerWithEmptyEnvironmentShouldFail")
+    public void testCreateHandlerWithEmptyEnvironmentShouldFail()  {
+        assertThrows(IllegalStateException.class, () -> new DynamoDBStreamHandler());
+    }
+
+
+
+    @Test
+    @DisplayName("testHandlerHandleSimpleModifyEventWithoutProblem")
+    public void testHandlerHandleSimpleModifyEventWithoutProblem() throws IOException {
+        DynamodbEvent requestEvent = loadEventFromResourceFile(SAMPLE_MODIFY_EVENT_FILENAME);
+        String response =  handler.handleRequest(requestEvent, context);
+        assertNotNull(response);
+    }
+
+    @Test
+    @DisplayName("testHandlerHandleSimpleInsertEventWithoutProblem")
+    public void testHandlerHandleSimpleInsertEventWithoutProblem() throws IOException {
+        DynamodbEvent requestEvent = loadEventFromResourceFile(SAMPLE_INSERT_EVENT_FILENAME);
+        String response =  handler.handleRequest(requestEvent, context);
+        assertNotNull(response);
+    }
+
+
+    @Test
+    @DisplayName("testHandlerHandleSimpleRemoveEventWithoutProblem")
+    public void testHandlerHandleSimpleRemoveEventWithoutProblem() throws IOException {
+        DynamodbEvent requestEvent = loadEventFromResourceFile(SAMPLE_REMOVE_EVENT_FILENAME);
+        String response =  handler.handleRequest(requestEvent, context);
+        assertNotNull(response);
+    }
+
+    @Test
+    @DisplayName("testHandleUnknownEventToGiveException")
+    public void testHandleUnknownEventToGiveException() throws IOException {
+        DynamodbEvent requestEvent = loadEventFromResourceFile(SAMPLE_UNKNOWN_EVENT_FILENAME);
+        assertThrows(RuntimeException.class, () -> handler.handleRequest(requestEvent, context));
+    }
+
+    @Test
+    @DisplayName("testHandleExceptionInEventHandlingShouldGiveException")
+    public void testHandleExceptionInEventHandlingShouldGiveException() throws IOException, InterruptedException {
+        DynamodbEvent requestEvent = loadEventFromResourceFile(SAMPLE_MODIFY_EVENT_FILENAME);
```

Comment:
Use nva commons TestAppedner for testing the logging of the Exception message. You might need hamcrest as well to check for the content of the message.

```
TestingAppender appender =LogUtils.getTestingAppender(ClassThatThrowsException.class);
assertThat(appender.getMessages(),containstString(expectedMessage));
....

Reply:
Use TestAppender and check that exceptions send correct messages and logger logs error. Use TestAppender